knowledge-base/adr/0031-resilience-options.md

```diff
@@ -249,78 +302,84 @@ function globalResilience(middleWare: Middleware, id: string) {}
-- Defaults non-breaking
-- TypeScript shows options
+type HttpRequestOptions = {
+    csrfTokenFetching:boolean,
+    resilience?: MiddleWare
+}
+```
 
-Contra:
+The resilience value could be a totally custom middleware, one with custom options or if not given the default.
+If methods execute http request they will get the additional option.
+For example the `getDestination()` or `serviceToken()` methods:
 
-- No custom implementation
-- We have to make assumption on reasonable behavior
+```ts
+function getDestination(
+    options: DestinationFetchOptions & DestinationForServiceBindingOptions,
+    httpRequestOptions?: HttpRequestOptions //contains resilience
+){
+  //implementation
+}
 
-#### API B - Middleware (Default off)
+function serviceToken( 
+    service: string | Service,        
+    options?: OldOptions & {resilience?: MiddleWare}
+){
+  //implementation
+}
+```
 
-- Users can pass a function taking a function with arguments and returning a `Promise<T>`.
-- For example the input could be the `http-request` function with arguments and the return the promise of `http-client`
-- We provide a sample implementation `resilience()` for easy consumption
-- Assumes all resilience is switched off per default
+### Deprecation Period
 
-```ts
-import {executeHttpRequest} from "@sap-cloud-sdk/http-client";
+We will have a deprecation period for the old options.
+If both options are given then new one should win.
 
+```ts
 myApi
   .getAll()
-  .middleware(resilience({retry: 3}))
-  .execute({destinationName: 'my-dest'});
+  .timeout(456)                //deprecated      
+  .middleware(resilience({ circuitBreaker: ()=>'disabled',timeout:()=>123 }), id)
+  .execute({ destinationName: 'my-dest',
+    timeout:456,                //deprecated      
+    disableCiruitBreaker:true   //deprecated      
+  });
+
+//user 123 as timeout and circuit breaker of
 
 executeHttpRequest({
-    destinationName: 'my-dest'
-  },
-  {
-    resilience: resilience({...})
-  }
+          destinationName: 'my-dest'
+        },
+        {
+          timeout: 456,  
+          resilience: resilience({ timeout: ()=>123 })
+        }
 );
 
-type Middleware<T> = <T>(
-        fn: () => Promise<T>,
-        context?: 'service' | 'target'
-) => Promise<T>;
+//use 123 as timeout
 ```
 
-Pro:
-
-- Easy to use
-- Flexible
-- TypeScript shows options
+In the intermediate period where options can be given via the old and new we have to do evaluation:
 
-Contra:
+```ts
+import {defaultResilienceBTPServices} from "@sap-cloud-sdk/connectivity/dist/scp-cf";
 
-- Extension of existing resilience not possible
+function getClientCredentialsToken(service: string | Service,
+                                   userJwt?: string | JwtPayload,
+                                   options?: ResilienceOptions & { resilience: ResilienceMiddleWareOptions }) {
+    
+   const timeout = options.resilience.timeout || options.timeout ? () => options.timeout : defaultResilienceOptions.timeout
```

Comment:
```suggestion
   const timeout = options.resilience.timeout || (options.timeout ? () => options.timeout : defaultResilienceOptions.timeout)
```

A pitfall that I fell into. 😆 

`true || false ? 1 : 0` equals `1` instead of `true`.